hub/src/main/java/ch/prevo/open/hub/nodes/NodeService.java

```diff
@@ -44,25 +51,61 @@ public NodeService(RestTemplateBuilder restTemplateBuilder) {
     }
 
     private List<InsurantInformation> lookupInsurantInformationList(String url) {
-        InsurantInformation[] nodeExits = restTemplate.getForObject(url, InsurantInformation[].class);
-        return nodeExits == null ? emptyList() : asList(nodeExits);
+        try {
+            InsurantInformation[] nodeExits = restTemplate.getForObject(url, InsurantInformation[].class);
+            return nodeExits == null ? emptyList() : asList(nodeExits);
+        } catch (Exception e) {
+            LOGGER.error("Could not fetch data from URL {}", url, e);
+        }
+        return emptyList();
     }
 
     public void notifyMatches(List<Match> matches) {
-        for (NodeConfiguration nodeConfig : nodeRegistry.currentNodes()) {
-            notifyMatches(nodeConfig, matches);
+        List<NodeConfiguration> nodeConfigurations = nodeRegistry.currentNodes();
+
+        for (Match match : matches) {
+            try {
+                Pair<NodeConfiguration, NodeConfiguration> nodesToNotify = findNodesToNotify(match, nodeConfigurations);
+
+                // notify previous node
+                tryNotifyMatch(nodesToNotify.getLeft(), match);
+                tryNotifyMatch(nodesToNotify.getRight(), match);
+            } catch (Exception e) {
+                LOGGER.error("Unexpected error occurred while notifying match: {}", match, e);
+            }
         }
     }
 
-    private void notifyMatches(NodeConfiguration nodeConfig, List<Match> matches) {
-        for (Match match : matches) {
-            if (nodeConfig.containsRetirementFundUid(match.getPreviousRetirementFundUid())
-                    || nodeConfig.containsRetirementFundUid(match.getNewRetirementFundUid())) {
-                MatchNotification matchNotification = new MatchNotification();
-                matchNotification.setEncryptedOasiNumber(match.getEncryptedOasiNumber());
-                matchNotification.setNewRetirementFundUid(match.getNewRetirementFundUid());
-                restTemplate.postForEntity(nodeConfig.getMatchNotifyUrl(), matchNotification, String.class);
+    private Pair<NodeConfiguration, NodeConfiguration> findNodesToNotify(Match match,
```

Comment:
when `lookupInsurantInformationLis` is changed to only accept valid UIDs, this method could be greatly simplified to fetch a single `NodeConfiguration`. This would in turn make `notifyMatches` simpler (no exception handling) and more readable (i.e., the semantic overhead of mapping 'left' to 'previous fund', 'right' to 'new fund'.

Reply:
done